lib/ansible/cli/doc.py

```diff
@@ -842,8 +842,9 @@ def format_plugin_doc(plugin, plugin_type, doc, plainexamples, returndocs, metad
         if context.CLIARGS['show_snippet']:
             if plugin_type not in SNIPPETS:
                 raise AnsibleError("Snippets are only available for the following plugin types: %s" % ', '.join(SNIPPETS))
-            if plugin_type == 'inventory' and plugin in ('yaml', 'toml'):
-                # not usable as other inventory plugins
+            if plugin_type == 'inventory' and doc.get('options') and not doc['options'].get('plugin'):
+                # these are 'configurable' but not intended for yaml type inventory sources, like ini or script
```

Comment:
`ini` is mentioned but doesn't get here since it doesn't have `options`. Maybe

```python
if plugin_type == 'inventory' and (not doc.get('options') or not doc['options'].get('plugin')):
```

Reply:
Oh, I guess we don't need to delete options in that case. Nevermind me :laughing:

LGTM

Reply:
eggzactly